tensorflow/compiler/mlir/xla/transforms/hlo_legalize_to_lhlo.cc

```diff
@@ -265,30 +256,131 @@ class HloToLhloReturnConverter : public OpConversionPattern<xla_hlo::ReturnOp> {
 //   return
 //  }
 // }
-struct HloLegalizeToLhlo : public FunctionPass<HloLegalizeToLhlo> {
-  void runOnFunction() override {
+struct HloLegalizeToLhlo : public ModulePass<HloLegalizeToLhlo> {
+  void runOnModule() override {
     OwningRewritePatternList patterns;
-    ConversionTarget target(getContext());
+    auto& context = getContext();
+    ConversionTarget target(context);
     target.addLegalDialect<xla_lhlo::XlaLhloDialect>();
+    target.addLegalDialect<StandardOpsDialect>();
+    target.addIllegalOp<mlir::ReturnOp>();
+    target.addIllegalOp<mlir::TensorLoadOp>();
+    target.addIllegalOp<mlir::TensorStoreOp>();
+    target.addLegalOp<ModuleTerminatorOp>();
+    target.addIllegalDialect<xla_hlo::XlaHloDialect>();
+    target.addDynamicallyLegalOp<FuncOp>([&](FuncOp op) {
+      bool result = true;
+      for (auto& input : op.getType().getInputs()) {
+        result &= input.isa<MemRefType>();
+      }
+      return result;
+    });
+
+    auto module = getModule();
+    populateHLOToLHLOConversionPattern(module.getContext(), &patterns);
 
-    auto func = getFunction();
-    populateHLOToLHLOConversionPattern(func.getContext(), &patterns);
-    if (failed(applyPartialConversion(func, target, patterns, nullptr))) {
+    SmallVector<Operation*, 16> ops;
+    for (auto& op : module.getBody()->getOperations()) {
+      ops.push_back(&op);
+    }
+    if (failed(applyFullConversion(ops, target, patterns, nullptr))) {
       signalPassFailure();
     }
   }
 };
 
+Type convertType(Type t) {
```

Comment:
Function naming convention (unless overloaded from base class) should follow https://google.github.io/styleguide/cppguide.html#Function_Names